components/brave_ads/core/internal/serving/notification_ad_serving.cc

```diff
@@ -110,7 +110,6 @@ void NotificationAdServing::BuildUserModelCallback(
     const UserModelInfo& user_model) {
   NotifyOpportunityAroseToServeNotificationAd(user_model.interest.segments);
 
-  CHECK(eligible_ads_);
```

Comment:
`CHECK` is not required because we call `IsSupported` within the flow of `MaybeServeAd`